Tasks/Src/Task_CheckContactor.c

```diff
@@ -1,16 +1,52 @@
 /* Copyright (c) 2018-2022 UT Longhorn Racing Solar */
-#include "RTOS_BPS.h"
+#include "config.h"
 #include "Contactor.h"
+#include "CANbus.h"
 #include "Tasks.h"
+#include "BSP_UART.h"
+#include "Voltage.h"
+#include "CAN_Queue.h"
 
-/* 
- * Launch this task once the contactor has been closed.
- * This will trip the BPS if the contactor is open.
- * The purpose of this is to trip the BPS if the emergency
- * cutoff switch is pressed.
- */
 void Task_CheckContactor(void *p_arg) {
     (void)p_arg;
+    
+    CANMSG_t CanMsg = {.payload = {.idx = 0, .data.b = 1}};
+
+    // launch watchdog task
+    RTOS_BPS_TaskCreate(&PetWDog_TCB,              // TCB
+                "TASK_PETWDOG_PRIO",        // Task Name (String)
+                Task_PetWDog,               // Task function pointer
+                (void *)0,                  // Task function args
+                TASK_PETWDOG_PRIO,          // Priority
+                PetWDog_Stk,                // Stack
+                TASK_PETWDOG_STACK_SIZE
+                );
+
+    // If a contactor is on before we turn it on in this task, it may have failed and welded shut
+    if (Contactor_GetState(HVHIGH_CONTACTOR) || Contactor_GetState(HVLOW_CONTACTOR)) {
+        Fault_BitMap |= Fault_ESTOP;
+        EnterFaultState();
+    } else {
+        RTOS_BPS_SemPost(&SafetyCheck_Sem4, OS_OPT_POST_1);
+    }
```

Comment:
What is the point of posting this semaphore if it is pended right after? I don't think it is necessary to do this and increase the value of NUM_FAULT_POINTS.

Reply:
to be fair num fault points would be a bit of a misnomer if it didn't include one of the fault points

Reply:
Lol, NUM_FAULT_POINTS includes open wire which we don't even check anymore. Ig it's fine to keep it. Just seems redundant imo.